packages/metrics/src/Metrics.ts

```diff
@@ -587,8 +599,14 @@ class Metrics extends Utility implements MetricsInterface {
           'If application metrics should never be empty, consider using `throwOnEmptyMetrics`'
       );
     }
-    const emfOutput = this.serializeMetrics();
-    hasMetrics && this.console.log(JSON.stringify(emfOutput));
+
+    if (!this.disabled) {
+      const emfOutput = this.serializeMetrics();
+      hasMetrics && this.console.log(JSON.stringify(emfOutput));
+    } else {
+      // TODO log warning that metrics are disabled?
+    }
```

Comment:
A few notes here:
- added the check on `this.disabled` here instead of in `addMetric` in case it should be possible to "toggle" between enabled and disabled in the future
- added an else-clause in case we need to log a warning if metrics are disabled, like with the empty metrics warning

Reply:
Thanks for the added details!

I agree with checking for the value here, at least for now this is the least disruptive place to make this change.

Regarding the `if/else` for now I think we can remove it and avoid logging any warning. This is an opt-in behavior that can be enabled only via environment variable, so I think it's fine to stay quiet. 

This is also a feature we recommend explicitly only for development, and for other utilities like Tracer we also don't emit any warning when disabled explicitly.